kernel/data-pipeline/api/src/main/java/org/apache/shardingsphere/data/pipeline/api/datasource/config/impl/ShardingSpherePipelineDataSourceConfiguration.java

```diff
@@ -88,18 +91,15 @@ private String getJdbcUrl(final Map<String, Object> props) {
     
     private void appendJdbcQueryProperties(final String databaseType) {
         Optional<JdbcQueryPropertiesExtension> extension = PipelineTypedSPILoader.findDatabaseTypedService(JdbcQueryPropertiesExtension.class, databaseType);
-        if (!extension.isPresent()) {
+        if (!extension.isPresent() || extension.get().extendQueryProperties(new Properties()).isEmpty()) {
             return;
         }
```

Comment:
Could we remove ` || extension.get().extendQueryProperties(new Properties()).isEmpty()`? There's another `extendQueryProperties` invocation